### PR TITLE
[NumPy] Updates for §Indexing

### PR DESCRIPTION
**Pull request by dkillick**
Updates to the section on Indexing, including ellipsis (a pre-requisite for the Iris course) and the concept of slicing. I've also slimmed one of the code examples and updated the section summary.

### CODE REVIEW

**Comment by dkillick** [comment]
@pp-mo you made some similar suggestions for this section, would you mind taking a look?

**Comment by dkillick** [comment]
Rebase to fix merge conflicts.

File: course_content/notebooks/numpy_intro.ipynb [hunk 1]
```diff
@@ -423,6 +428,49 @@
...
+   "outputs": [],
+   "source": [
+    "print('Second row:  {} is equivalent to {}'.format(arr_2d[1], arr_2d[1, :]))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This is known as **ellipsis**. Ellipsis can be specified explicitly using \"`...`\", which automatically expands to \"`:`\" for each dimension unspecified in the slice:"
```
Comment (pp-mo) [inline course_content/notebooks/numpy_intro.ipynb]: This + following examples are nice : show ellipsis at start + in middle.
I'm wondering : Is this a place to explain that "any unspecified (remaining) indices are filled with ':', so trailing ellipses or colons are simply redundant ".

**Comment by dkillick** [comment]
Thanks @pp-mo 👍

**Pull request merged**